src/test/expression/expression_evaluator_to_pos_list_test.cpp

```diff
@@ -99,6 +108,12 @@ TEST_F(ExpressionEvaluatorToPosListTest, PredicatesWithOnlyLiterals) {
   EXPECT_TRUE(test_expression(table_b, ChunkID{1}, *greater_than_(5, 1), {0, 1, 2}));
   EXPECT_TRUE(test_expression(table_b, ChunkID{1}, *between_(2, 5, 6), {}));
   EXPECT_TRUE(test_expression(table_b, ChunkID{0}, *between_(2, 1, 6), {0, 1, 2, 3}));
+  EXPECT_TRUE(test_expression(table_b, ChunkID{1}, *between_lower_exclusive_(2, 5, 6), {}));
+  EXPECT_TRUE(test_expression(table_b, ChunkID{0}, *between_lower_exclusive_(2, 1, 6), {0, 1, 2, 3}));
```

Comment:
the tests you added would pass even if all between types were rewritten to the same expression. Please write tests that actually test the lower and upper bound.